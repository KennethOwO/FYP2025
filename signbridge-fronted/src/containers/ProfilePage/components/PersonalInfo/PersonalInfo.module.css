.personalInfoContainer{
    display: flex;
    flex-direction: column;
    /* align-items: center; */
    /* justify-content: center; */
    margin-top: 100px;
    margin-bottom: 20px;
}

.personalInfoBox{
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    padding: 40px;
    border-radius: 10px;
    background-color: white;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    transition: box-shadow 0.3s ease-in-out;
}

.profilePicture{
    width: 100px;
    height: 100px;
    border-radius: 50%;
    margin-bottom: 20px;
    padding: 5px;
    border: black 1px solid;
    object-fit: cover;
    object-position: center;
}

.email {
    font-size: 20px;
    margin: 10px 0;
}

.editButton {
    margin-top: 20px;
    padding: 10px 20px;
    border-radius: 5px;
    background-color: #f8e2cd;
    color: #8d4e13;
    font-size: 16px;
    cursor: pointer;
    transition: background-color 0.3s ease-in-out;
    border: none;
}

.editButton:hover {
    background-color: #f0d3b9;
}

.joined {
    margin-top: 40px;
    font-weight: 400;
    color: #8d4e1393;
}

@media screen and (max-width: 1100px) {
    .personalInfoBox {
        padding: 20px;
    }

    .profilePicture {
        width: 80px;
        height: 80px;
        margin-bottom: 15px;
    }

    .name {
        font-size: 24px;
    }

    .email {
        font-size: 16px;
    }

    .editButton {
        padding: 8px 16px;
        font-size: 14px;
        margin-top: 15px;
    }

    .joined {
        margin-top: 15px;
    }
}

@media screen and (max-width: 850px) {
    .personalInfoContainer {
        margin-top: 50px;
    }

    .profilePicture {
        width: 100px;
        height: 100px;
        margin-bottom: 10px;
    }

    .name {
        font-size: 32px;
    }

    .email {
        font-size: 20px;
    }

    .editButton {
        padding: 6px 12px;
        font-size: 12px;
        margin-top: 10px;
    }

    .joined {
        margin-top: 20px;
    }
}