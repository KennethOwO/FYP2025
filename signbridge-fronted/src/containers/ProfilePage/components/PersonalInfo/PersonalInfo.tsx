import { useTranslation } from "react-i18next";
import style from "./PersonalInfo.module.css";

import { useUserStore } from "@root/store/userStore";

const PersonalInfo = () => {
	const { t } = useTranslation();
	const { user } = useUserStore();

	// "Sat, 07 Sep 2024 08:44:21 GMT" => "Sep, 2024"
	const formatDate = (dateString: string) => {
		const date = new Date(dateString);
		const options: Intl.DateTimeFormatOptions = {
			year: "numeric",
			month: "short",
		};
		const formattedDate = date.toLocaleDateString("en-US", options);

		const [month, year] = formattedDate.split(" ");
		return `${month}, ${year}`;
	};

	return (
		<div className={style.personalInfoContainer}>
			<div className={style.personalInfoBox}>
				<img src={user?.picture} alt="Profile Picture" className={style.profilePicture} />
				<div className={style.personalInfo}>
					<h1 className={style.name}>{user?.username}</h1>
					<p className={style.email}>{user?.email}</p>
					{/* <button className={style.editButton}>Edit Profile</button> */}
					<p className={style.joined}>
						{t("joinedOn")} {formatDate(user?.created_at ?? "")}
					</p>
				</div>
			</div>
		</div>
	);
};

export default PersonalInfo;
