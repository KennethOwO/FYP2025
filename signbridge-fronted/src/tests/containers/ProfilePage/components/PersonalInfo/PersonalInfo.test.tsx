import PersonalInfo from "@root/containers/ProfilePage/components/PersonalInfo/PersonalInfo";
import { create } from "react-test-renderer";
import { cleanup } from "@testing-library/react";
jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock the translation function to return the key
    i18n: {
      changeLanguage: jest.fn(), // Mock the changeLanguage function
    },
  }),
}));
// Mock Cookies.get() before your test cases
jest.mock("js-cookie", () => ({
  get: jest.fn(),
}));

// or if you have the get method implementation
jest.mock("js-cookie");

describe("Test PersonalInfo", () => {
  it("should render correctly", () => {
    const tree = create(<PersonalInfo />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
