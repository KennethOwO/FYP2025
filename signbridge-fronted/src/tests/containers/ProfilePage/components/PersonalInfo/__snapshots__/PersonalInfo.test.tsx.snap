// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PersonalInfo should render correctly 1`] = `
<div>
  <div>
    <img
      alt="Profile Picture"
    />
    <div>
      <h1 />
      <p />
      <p>
        joinedOn
         
        Invalid, Date
      </p>
    </div>
  </div>
</div>
`;
